### PR TITLE
Fix typos, repeated words, align spelling to LDP

### PR DESCRIPTION
**Pull request by gustafn**
This is a follow-up commit to https://github.com/openssl/openssl/pull/12320 and https://github.com/openssl/openssl/pull/12460 and concerns all the .c- and .h-files of the project. The discussion in https://github.com/openssl/openssl/pull/12320 showed interest in source code changes, which are provided by this pull request. The changes are relative to the master branch of GitHub from Thu Jul 23 17:40:40 2020.

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

File: crypto/dsa/dsa_local.h [hunk 1]
```diff
@@ -14,7 +14,7 @@
 struct dsa_st {
     /*
      * This first variable is used to pick up errors where a DSA is passed
-     * instead of of a EVP_PKEY
+     * instead of a EVP_PKEY
```
Comment (paulidale) [inline crypto/dsa/dsa_local.h:17]: _an_ instead of _a_ here?

File: crypto/modes/ocb128.c [hunk 2]
```diff
@@ -101,7 +101,7 @@ static OCB_BLOCK *ocb_lookup_l(OCB128_CONTEXT *ctx, size_t idx)
     if (idx >= ctx->max_l_index) {
         void *tmp_ptr;
         /*
-         * Each additional entry allows to process almost double as
+         * Each additional entry allows one to process almost double as
```
Comment (paulidale) [inline crypto/modes/ocb128.c:104]: Should _double_ be _twice_?

File: crypto/ts/ts_err.c [hunk 3]
```diff
@@ -39,7 +39,7 @@ static const ERR_STRING_DATA TS_str_reasons[] = {
     {ERR_PACK(ERR_LIB_TS, 0, TS_R_NONCE_MISMATCH), "nonce mismatch"},
     {ERR_PACK(ERR_LIB_TS, 0, TS_R_NONCE_NOT_RETURNED), "nonce not returned"},
     {ERR_PACK(ERR_LIB_TS, 0, TS_R_NO_CONTENT), "no content"},
-    {ERR_PACK(ERR_LIB_TS, 0, TS_R_NO_TIME_STAMP_TOKEN), "no time stamp token"},
+    {ERR_PACK(ERR_LIB_TS, 0, TS_R_NO_TIME_STAMP_TOKEN), "no timestamp token"},
```
Comment (paulidale) [inline crypto/ts/ts_err.c:42]: The various TIME_STAMP macros should be updated in due course.  It doesn't have to be in this PR.

File: include/crypto/bn.h [hunk 4]
```diff
@@ -61,7 +61,7 @@ void bn_set_static_words(BIGNUM *a, const BN_ULONG *words, int size);
 int bn_set_words(BIGNUM *a, const BN_ULONG *words, int num_words);
 
 /*
- * Some BIGNUM functions assume most significant limb to be non-zero, which
+ * Some BIGNUM functions assume most significant limb to be nonzero, which
```
Comment (paulidale) [inline include/crypto/bn.h:64]: Shouldn't it be: _assume **the** most significant limb to be_

File: include/internal/ktls.h [hunk 5]
```diff
@@ -102,7 +102,7 @@ static ossl_inline int ktls_read_record(int fd, void *data, size_t length)
 }
 
 /*
- * KTLS enables the sendfile system call to send data from a file over
+ * KTLS enables the sendfilesystem call to send data from a file over
```
Comment (paulidale) [inline include/internal/ktls.h:105]: This is wrong: _sendfile system call_ or even _sendfile(2) system call_.

File: include/internal/ktls.h [hunk 6]
```diff
@@ -237,7 +237,7 @@ static ossl_inline int ktls_send_ctrl_message(int fd, unsigned char record_type,
 }
 
 /*
- * KTLS enables the sendfile system call to send data from a file over TLS.
+ * KTLS enables the sendfilesystem call to send data from a file over TLS.
```
Comment (paulidale) [inline include/internal/ktls.h:240]: This is wrong: _sendfile system call_ or even _sendfile(2) system call_.

File: include/internal/packet.h [hunk 7]
```diff
@@ -633,7 +633,7 @@ struct wpacket_st {
     /* Number of bytes written so far */
     size_t written;
 
-    /* Maximum number of bytes we will allow to be written to this WPACKET */
+    /* Maximum number of bytes we will allow one to be written to this WPACKET */
```
Comment (paulidale) [inline include/internal/packet.h:636]: This looks like an incorrect change.  _Maximum number of allowed bytes to write to this WPACKET_ ?

File: include/openssl/asn1.h [hunk 8]
```diff
@@ -217,7 +217,7 @@ typedef struct ASN1_TLC_st ASN1_TLC;
 /* This is just an opaque pointer */
 typedef struct ASN1_VALUE_st ASN1_VALUE;
 
-/* Declare ASN1 functions: the implement macro in in asn1t.h */
+/* Declare ASN1 functions: the implement macroin asn1t.h */
```
Comment (paulidale) [inline include/openssl/asn1.h:220]: This should be _macro in_.  I.e. delete one _in_ but not the extra space.

File: crypto/cmp/cmp_local.h [hunk 9]
```diff
@@ -33,7 +33,7 @@ struct ossl_cmp_ctx_st {
 
     /* message transfer */
     OSSL_CMP_transfer_cb_t transfer_cb; /* default: OSSL_CMP_MSG_http_perform */
-    void *transfer_cb_arg; /* allows to store optional argument to cb */
+    void *transfer_cb_arg; /* allows one to store optional argument to cb */
```
Comment (paulidale) [inline crypto/cmp/cmp_local.h:36]: What about: _allows one to store an optional argument_ ?

File: crypto/cmp/cmp_local.h [hunk 10]
```diff
@@ -45,7 +45,7 @@ struct ossl_cmp_ctx_st {
     /* attempts polling for a response if a 'waiting' PKIStatus is received */
     time_t end_time; /* session start time + totaltimeout */
     OSSL_HTTP_bio_cb_t http_cb;
-    void *http_cb_arg; /* allows to store optional argument to cb */
+    void *http_cb_arg; /* allows one to store optional argument to cb */
```
Comment (paulidale) [inline crypto/cmp/cmp_local.h:48]: Perhaps instead: _allows one to store an optional argument_ ?

File: crypto/dsa/dsa_local.h (hunk 1 above)
Comment (gustafn) [reply crypto/dsa/dsa_local.h:17]: of course. there are several more of these that i'll address in the update

File: include/crypto/bn.h (hunk 4 above)
Comment (gustafn) [reply include/crypto/bn.h:64]: right. the same wording was used as well in crypto/bn/bn_div.c

File: crypto/asn1/a_int.c [hunk 11]
```diff
@@ -54,15 +54,15 @@ int ASN1_INTEGER_cmp(const ASN1_INTEGER *x, const ASN1_INTEGER *y)
...
  * complement and 0 by the add one (due to carry) so just copy as many trailing
  * zeros to the destination as there are in the source. The carry will add one
  * to the last none zero octet: so complement this octet and add one and finally
  * complement any left over until you get to the start of the string.
  *
  * Padding is a little trickier too. If the first bytes is > 0x80 then we pad
- * with 0xff. However if the first byte is 0x80 and one of the following bytes
- * is non-zero we pad with 0xff. The reason for this distinction is that 0x80
+ * with 0xff. However, if the first byte is 0x80 and one of the following bytes
+ * is nonzero we pad with 0xff. The reason for this distinction is that 0x80
```
Comment (levitte) [inline crypto/asn1/a_int.c:65]: ?

File: crypto/asn1/a_int.c [hunk 12]
```diff
@@ -54,15 +54,15 @@ int ASN1_INTEGER_cmp(const ASN1_INTEGER *x, const ASN1_INTEGER *y)
  * The DER representation of negative integers is in 2s complement form.
  * The internal form is converted by complementing each octet and finally
  * adding one to the result. This can be done less messily with a little trick.
- * If the internal form has trailing zeroes then they will become FF by the
+ * If the internal form has trailing zeros then they will become FF by the
```
Comment (levitte) [inline crypto/asn1/a_int.c:57]: ?

File: apps/cmp.c [hunk 13]
```diff
@@ -2889,7 +2889,7 @@ int cmp_main(int argc, char **argv)
 
     /* read default values for options from config file */
     configfile = opt_config != NULL ? opt_config : default_config_file;
-    if (configfile && configfile[0] != '\0' /* non-empty string */
+    if (configfile && configfile[0] != '\0' /* nonempty string */
```
[the remark this answers is not in the document]
Comment (gustafn) [reply apps/cmp.c:2878]: These changes are applying the recommended rules of the Linux Documentation Project https://man7.org/linux/man-pages/man7/man-pages.7.html

These changes are as well included in openssl/util/find-doc-nits

File: crypto/asn1/a_int.c (hunk 12 above)
Comment (paulidale) [reply crypto/asn1/a_int.c:57]: Zeros is plural and zeroes is a verb.

**Comment by gustafn** [comment]
To address the comment "US vs UK english stuff?" directly: 
as discussed in https://github.com/openssl/openssl/pull/12299 i've left out any changes addressing the "US vs. UK" question. The code base is a mixture of the US and UK spelling conventions, but closer to US.

**Comment by gustafn** [comment]
Hmm, i've resolved the conflicts of this pull requests by merging with master (The conflicts were introduced by the master branch evolving). But now, all the changes that happened in the meantime are showing up as changes in my last commit.  i should probably undo my merge.

**Comment by t8m** [comment]
Please do not do merge commits as they are not allowed. Could you please git rebase instead?

**Comment by gustafn** [comment]
ok. i'll undo make an rebase attempt.

**Comment by t8m** [comment]
There are still 2 merge commits.

**Comment by gustafn** [comment]
These merges were the result of the resolving the conflicts (on a "git pull upstream"). So, i have probably reset everything to my starting point from master, update then master (should be free of conflicts) and trying the reapply my changes afterwards. Probably, one gets as well merges, when doing a "Resolve conflicts" via the GitHub interface.

**Comment by t8m** [comment]
Yes, unfortunately resolving conflicts via the GitHub interface cannot be done.

**Comment by t8m** [comment]
Could you please rebase if you're still interested?

**Comment by gustafn** [comment]
i will try to revamp it, but i will take me a few days...

**Comment by gustafn** [comment]
It seems that the "push force" i used to get this voluminous PR in sync again closed the PR. The new one is in #16712

**Pull request closed without merging**